flash/data/data_module.py

```diff
@@ -170,15 +170,15 @@ def _show_batch(self, stage: RunningStage, func_names: Union[str, List[str]], re
 
         iter_dataloader = getattr(self, iter_name)
         with self.data_fetcher.enable():
+            if reset:
+                self.data_fetcher.batches[stage] = {}
             try:
                 _ = next(iter_dataloader)
             except StopIteration:
                 iter_dataloader = self._reset_iterator(stage)
                 _ = next(iter_dataloader)
             data_fetcher: BaseVisualization = self.data_fetcher
             data_fetcher._show(stage, func_names)
-            if reset:
```

Comment:
Oh I see. The bug was introduced in 1 refactor. Better to reset data fetcher directly. Good catch !